mipengine/controller/api/validator.py

```diff
@@ -0,0 +1,378 @@
+import logging
+import traceback
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from mipengine.common.common_data_elements import CommonDataElement
+from mipengine.common.common_data_elements import common_data_elements
+from mipengine.common.node_catalog import node_catalog
+from mipengine.controller.algorithms_specifications import AlgorithmSpecifications
+from mipengine.controller.algorithms_specifications import (
+    CROSSVALIDATION_ALGORITHM_NAME,
+)
+from mipengine.controller.algorithms_specifications import GenericParameterSpecification
+from mipengine.controller.algorithms_specifications import InputDataSpecification
+from mipengine.controller.algorithms_specifications import InputDataSpecifications
+from mipengine.controller.algorithms_specifications import algorithms_specifications
+from mipengine.controller.api.AlgorithmRequestDTO import AlgorithmInputDataDTO
+from mipengine.controller.api.AlgorithmRequestDTO import AlgorithmRequestDTO
+from mipengine.controller.api.exceptions import BadRequest
+from mipengine.controller.api.exceptions import BadUserInput
+
+
+def validate_algorithm_request(algorithm_name: str, request_body: str):
+    """
+    Validates the proper usage of the algorithm:
+    1) algorithm exists,
+    2) algorithm body has proper format and
+    3) algorithm input matches the algorithm specifications.
+    """
+
+    if algorithm_name not in algorithms_specifications.enabled_algorithms.keys():
+        raise BadRequest(f"Algorithm '{algorithm_name}' does not exist.")
```

Comment:
Isnt there a NotFound in python?
Its more appropriate for the case.

Reply:
Good point but in this case all exceptions from the validator are either `BadRequest` or `BadUserInput`. Specific exceptions with specific behavior each one.
These are used to send the proper error response to the user.